src/Management/CodeGenerator/BaseCodeGenerator.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * This file is part of the contentful-management.php package.
+ *
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Management\CodeGenerator;
+
+use Contentful\Management\Resource\ContentType\Field\FieldInterface;
+use PhpParser\Comment;
+use PhpParser\Node;
+use PhpParser\PrettyPrinter\Standard as StandardPrettyPrinter;
+
+/**
+ * BaseCodeGenerator class.
+ */
+abstract class BaseCodeGenerator
+{
+    /**
+     * @var string
+     */
+    protected $defaultLocale;
+
+    /**
+     * BaseCodeGenerator constructor.
+     *
+     * @param string $defaultLocale
+     */
+    public function __construct(string $defaultLocale)
+    {
+        $this->defaultLocale = $defaultLocale;
+    }
+
+    /**
+     * @param array $params
+     */
+    abstract public function generate(array $params): string;
+
+    /**
+     * Returns a rendered node.
+     *
+     * @param Node $node
+     *
+     * @return string
+     */
+    protected function render(Node $node): string
+    {
+        $prettyPrinter = new StandardPrettyPrinter([
+            'shortArraySyntax' => true,
+        ]);
+
+        $code = $prettyPrinter->prettyPrintFile([$node]);
+
+        // Removes spaces from blank lines
+        $code = \preg_replace('/\n(\s+)\n/', "\n\n", $code)."\n";
+        // Removes space after parenthesis with return types
+        $code = \strtr($code, [') : ' => '): ']);
+
+        return $code;
+    }
+
+    /**
+     * Converts a string to SutdlyCaps.
+     *
+     * @param string $name
+     *
+     * @return string
+     */
+    protected function convertToStudlyCaps(string $name): string
+    {
+        return \ucwords(\str_replace(['-', '_'], ' ', $name));
+    }
+
+    /**
+     * @param (string|null)[] $classes
+     *
+     * @return Node\Stmt\Use_[]
+     */
+    protected function generateUses(array $classes): array
+    {
+        $classes = \array_filter($classes);
+
+        \usort($classes, function (string $classA, string $classB): int {
+            return $classA <=> $classB;
+        });
```

Comment:
Is this sorting them alphabetically? Is there any advantage to doing this?

Reply:
No real advantage, it's just to align to the coding standards used in the SDK. The code linter runs also on the fixtures directory, so I have a reference in terms of code quality.